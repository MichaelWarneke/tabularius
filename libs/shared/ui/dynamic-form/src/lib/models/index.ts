import {
  FormControl,
  ValidatorFn,
  AbstractControlOptions,
  AsyncValidatorFn,
  FormArray
} from '@angular/forms';

export class FormControlBase extends FormControl {
  controlType = '';
  key = '';
  label = '';
  maxWidth = 100;
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlTextbox extends FormControlBase {
  controlType = 'input';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlDate extends FormControlBase {
  controlType = 'datepicker';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlCheckbox extends FormControlBase {
  controlType = 'checkbox';
  type = '';
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlSelect extends FormControlBase {
  controlType = 'select';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}

export class FormControlRadioButton extends FormControlBase {
  controlType = 'radio-button';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export class FormControlArray extends FormArray {
  controlType = 'array';
  key = '';
  label = '';
  type = '';
  options?: string[];
  constructor(
    formState?: any,
    validatorOrOpts?:
      | ValidatorFn
      | ValidatorFn[]
      | AbstractControlOptions
      | null,
    asyncValidator?: AsyncValidatorFn | AsyncValidatorFn[] | null
  ) {
    super(formState, validatorOrOpts, asyncValidator);
  }
}
export { Validators } from '@angular/forms';
