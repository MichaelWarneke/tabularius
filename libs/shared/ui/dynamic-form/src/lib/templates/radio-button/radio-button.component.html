<div class="tabu-radio-button-container" [formGroup]="group">
  <label class="tabu-radio-label">{{ val?.label }}:</label>
  <mat-radio-group [formControlName]="val?.key">
    <mat-radio-button
      class="tabu-radio-button"
      *ngFor="let item of val?.options"
      [value]="item"
      >{{ item }}</mat-radio-button
    >
  </mat-radio-group>
</div>
