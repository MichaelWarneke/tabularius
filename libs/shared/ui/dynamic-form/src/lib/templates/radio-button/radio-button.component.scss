.tabu-radio-button-container {
  margin: 10px;
}

.tabu-radio-label {
  padding-right: 10px;
}

.tabu-radio-button {
  padding-right: 10px;
}
