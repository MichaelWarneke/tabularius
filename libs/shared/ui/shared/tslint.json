{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
