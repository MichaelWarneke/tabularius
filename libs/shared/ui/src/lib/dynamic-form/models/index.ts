export { QuestionBase } from './question-base';
export { DropdownQuestion } from './question-dropdown';
export { TextboxQuestion } from './question-textbox';
export { DatepickerQuestion } from './question-datepicker';
