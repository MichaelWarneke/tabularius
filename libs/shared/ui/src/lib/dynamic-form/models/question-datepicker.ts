import { QuestionBase } from './question-base';

export class DatepickerQuestion extends QuestionBase<string> {
  controlType = 'datepicker';
  type: Date;

  constructor(options: any = {}) {
    super(options);
    this.type = options['type'] || '';
  }
}
