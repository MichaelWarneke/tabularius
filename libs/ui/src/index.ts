export { TabuAppSidenavModule } from './lib/nav';
