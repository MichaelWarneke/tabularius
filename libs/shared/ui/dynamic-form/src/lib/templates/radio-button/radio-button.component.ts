import { Component, OnInit } from '@angular/core';
import { FormGroup } from '@angular/forms';
import { FormControlRadioButton } from '../../models';

@Component({
  selector: 'tabu-dynamic-form-radio-button',
  templateUrl: './radio-button.component.html',
  styleUrls: ['./radio-button.component.scss']
})
export class RadioButtonComponent {
  control: FormControlRadioButton | null = null;
  group: FormGroup | null = null;

  constructor() {}
}
