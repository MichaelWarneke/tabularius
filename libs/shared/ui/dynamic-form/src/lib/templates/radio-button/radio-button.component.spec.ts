import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { RadioButtonComponent } from './radio-button.component';
import { SharedModule } from '@tabularius/shared/ui';
import { FormGroup, FormControl } from '@angular/forms';
import { FormControlRadioButton } from '../../models';
import { By } from '@angular/platform-browser';

describe('RadioButtonComponent', () => {
  let component: RadioButtonComponent;
  let fixture: ComponentFixture<RadioButtonComponent>;
  const key = 'testKey';
  const label = 'testLabel';

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [RadioButtonComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(RadioButtonComponent);
    component = fixture.componentInstance;
    component.group = new FormGroup({ testKey: new FormControl('') });
    component.control = new FormControlRadioButton();
    component.control.key = key;
    component.control.label = label;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain the label', () => {
    const text = fixture.debugElement.query(By.css('label')).nativeElement
      .innerHTML;
    expect(text).toContain(label);
  });

  it('should contain the formControlName in attributes', () => {
    const attributes = fixture.debugElement.query(By.css('mat-radio-group'))
      .attributes;
    const text = attributes['ng-reflect-name'];
    expect(text).toContain(key);
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });
});
