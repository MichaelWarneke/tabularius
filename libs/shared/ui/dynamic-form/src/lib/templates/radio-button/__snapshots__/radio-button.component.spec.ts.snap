// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonComponent should render 1`] = `
<tabu-dynamic-form-radio-button
  control={[Function FormControlRadioButton]}
  group={[Function FormGroup]}
>
  <div
    class="tabu-radio-button-container ng-untouched ng-pristine ng-valid"
    ng-reflect-form="[object Object]"
  >
    <label
      class="tabu-radio-label"
    >
      testLabel:
    </label>
    <mat-radio-group
      class="mat-radio-group ng-untouched ng-pristine ng-valid"
      ng-reflect-name="testKey"
      role="radiogroup"
    >
      
    </mat-radio-group>
  </div>
</tabu-dynamic-form-radio-button>
`;
